src/Microsoft.VisualStudio.Debugger.Interop.UnixPortSupplier/Microsoft.VisualStudio.Debugger.Interop.UnixPortSupplier.cs

```diff
@@ -62,11 +62,24 @@ public interface IDebugUnixShellPort
         /// <returns>Home directory of the user.</returns>
         string GetUserHomeDirectory();
 
+        /// <summary>
+        /// Attaches gdbserver to a process.
+        /// </summary>
+        /// <param name="processId">Id of the process.</param>
+        /// <param name="preAttachCommand">Command to run before starting gdbserver.</param>
+        /// <returns>Communications addr:port</returns>
+        string AttachToProcess(int processId, string preAttachCommand);
```

Comment:
>AttachToProcess [](start = 15, length = 15)

This is a COM interface, so if we are going to add new things it should be in a new interface.

Reply:
fixed